test/unit/BondingCheckpoints.js

```diff
@@ -0,0 +1,1068 @@
+import Fixture from "./helpers/Fixture"
+import {functionSig} from "../../utils/helpers"
+import {assert} from "chai"
+import {ethers, web3} from "hardhat"
+import chai from "chai"
+import {solidity} from "ethereum-waffle"
+import {constants} from "ethers"
+
+chai.use(solidity)
+const {expect} = chai
+
+// TODO: to be moved in a separate util/config (i.e: chai-setup)
+chai.use(function(chai) {
+    const Assertion = chai.Assertion
+
+    Assertion.addMethod("matchStruct", function(expected) {
+        // eslint-disable-next-line no-invalid-this
+        const obj = this._obj
+
+        Object.keys(expected).forEach(function(key) {
+            if (Array.isArray(obj[key])) {
+                new Assertion(obj[key]).to.deep.eq(expected[key])
+            } else {
+                new Assertion(obj[key]).to.eq(expected[key])
+            }
+        })
+    })
+})
+
+describe("BondingCheckpoints", () => {
```

Comment:
Realized that these are currently a mix of unit and integration tests. Will be moving and refactoring things around for more well-defined tests